packages/ncids-js/src/components/nci-header/utils/megamenu.ts

```diff
@@ -0,0 +1,557 @@
+			sectionLink.innerHTML = `Explore ${parentLabel}`;
+			// append link into header
+			sectionHeader.append(sectionLink);
+			gridColHeader.append(sectionHeader);
+			// col for menu sections
+			const gridColMenu = this.createDom('div', ['usa-col', 'grid-gap-9']);
+
+			// make inside of mega menu async call
+			// pass in element that will be filled
+			this.getMenuLayer(gridColMenu, id);
+
+			gridRow.append(gridColHeader);
+			gridRow.append(gridColMenu);
+
+			// Append into parent Row section for Menus
+			menu.append(gridRow);
+			section.append(menu);
+			//return menu;
+		});
+	}
+
+	/**
+	 * Get the navigation data for the specific menu layer of
+	 * primary navigation navigation
+	 *
+	 * @param {Element} container Parent LI HTMLElement
+	 * @param {string} id The id to use for the api call
+	 *
+	 * @returns {Promise<void>}
+	 * @private
+	 */
+	private async getMenuLayer(container: Element, id: string): Promise<void> {
+		const data = this.getMenuData(id);
+		data.then((response) => {
+			const navdata: NavItem[] = [];
+			response.map((data) => {
+				navdata.push(data);
+			});
+			const card = this.createMenuLayer(navdata);
+			container.append(card);
+		});
+	}
+
+	/**
+	 * Create the Menu Layer and inject into primary
+	 * navigation
+	 *
+	 * @param {NavItem[]} data NavItem
+	 *
+	 * @return {Element} DIV layer HTML element
+	 * @private
+	 */
+	private createMenuLayer(data: NavItem[]): Element {
+		const amount = data.length || 0;
+		// determine if there is space left over in the menu
+		const leftover = amount - Math.floor(amount / 3) * 3;
+		const wrapper = this.createDom('div', ['wrapper']);
+		// Each set of 3 UL's are kept in one grid-row
+		let gridRowMenu = this.createDom('div', ['grid-row']);
+
+		// Go though each item in the response
+		[...data].map((item, index) => {
+			const mod = (index + 1) % 3;
+
+			// create the UL and container DIV
+			const MenuCol = this.createDom('div', ['usa-col', 'grid-gap-3']);
+			const MenuList = this.createDom('ul', ['nci-header-megamenu__list']);
+
+			// First link in list is Header link
+			const ParentMenuItem = this.createMenuLink(item.path, item.label);
+			MenuList.append(ParentMenuItem);
+			MenuCol.append(MenuList);
+			// Iterate over child items
+			[...item.children].map((item) => {
+				const MenuItem = this.createMenuLink(item.path, item.label);
+				MenuList.append(MenuItem);
+				MenuCol.append(MenuList);
+			});
+			gridRowMenu.append(MenuCol);
+			wrapper.append(gridRowMenu);
+			// if zero then we're at 3 and we should make a new row
+			if (mod === 0) {
+				gridRowMenu = this.createDom('div', ['grid-row']);
+			}
+		});
+
+		// If there is a leftover space then we need to fill
+		// it. Based on the amount of items in data, create
+		// an empty div and append to the last row container
+		if (leftover > 0) {
+			const MenuFiller = this.createDom('div', [
+				'usa-col',
+				leftover < 2 ? 'grid-gap-6' : 'grid-gap-3',
```

Comment:
Every if statement will trigger a need for code coverage. You'll need to make a test for both of these scenarios.